crypto/src/note.rs

```diff
@@ -236,6 +236,27 @@ impl Note {
     }
 }
 
+/// Create a note commitment from its parts.
+pub fn commitment(
+    note_blinding: Fq,
+    value: Value,
+    diversified_generator: decaf377::Element,
+    transmission_key_s: Fq,
+) -> Commitment {
+    let commit = poseidon377::hash_5(
+        &NOTECOMMIT_DOMAIN_SEP,
+        (
+            note_blinding,
+            value.amount.into(),
+            value.asset_id.0,
+            diversified_generator.compress_to_field(),
+            transmission_key_s,
+        ),
+    );
+
+    Commitment(commit)
+}
+
```

Comment:
Should this be a method on `Note`?

Reply:
i.e., do we have a reason to construct a note commitment from raw parts without constructing a note?

Reply:
We do construct a commitment this way without starting from a note, in two places, where we verify spend and output proofs, respectively:

https://github.com/penumbra-zone/penumbra/blob/e45943eb45ba4f22575b18a3b0d6726a276d3723/crypto/src/proofs/transparent.rs#L82-L87

https://github.com/penumbra-zone/penumbra/blob/e45943eb45ba4f22575b18a3b0d6726a276d3723/crypto/src/proofs/transparent.rs#L172-L177

Is there a better way to do this? I also agree it's a little awkward, but I didn't feel like a static method on `Note` would make more sense than this.

Reply:
The alternative would be to construct a `Note` out of the parts, and then call `note.commit` on the resulting instance.